tests/cython/PythonEvalUTest.cxxtest

```diff
@@ -136,81 +136,59 @@ public:
+        PythonEval::create_singleton_instance();
 
         // Delete the singleton instance of the PythonEval.
         PythonEval::delete_singleton_instance();
 
-
-        // Create an atomspace.
-        atomSpace = new AtomSpace();
-
-        // Tell the python evaluator to create its singleton instance
-        // with this atomspace.
-        PythonEval::create_singleton_instance(atomSpace);
+        PythonEval::create_singleton_instance();
 
         // Delete the singleton instance of the PythonEval.
         PythonEval::delete_singleton_instance();
 
-        // Delete the atomspace.
-        delete atomSpace;
-        atomSpace = NULL;
 
         // Initialize Python.
         global_python_initialize();
 
-        // Create an atomspace.
-        atomSpace = new AtomSpace();
-
-        // Tell the python evaluator to create its singleton instance
-        // with this atomspace.
-        PythonEval::create_singleton_instance(atomSpace);
+        // Repeat the process
+        PythonEval::create_singleton_instance();
 
         // Delete the singleton instance of the PythonEval.
         PythonEval::delete_singleton_instance();
 
-        // Delete the atomspace.
-        delete atomSpace;
-        atomSpace = NULL;
-
         logger().debug("[PythonEvalUTest] testPythonEvalSingletonCreationDeletion() DONE");
     }
 
     void testPythonEvalEvalExpr()
     {
         logger().debug("[PythonEvalUTest] testPythonEvalEvalExpr()");
-
-        // Create an atomspace and singleton instance to use it.
-        AtomSpace* atomSpace = new AtomSpace();
-        PythonEval::create_singleton_instance(atomSpace);
+        PythonEval::create_singleton_instance();
         PythonEval* evaluator = &PythonEval::instance();
 
         // Check that our atom space global is visible and can add a node.
         evaluator->clear_pending();
         evaluator->eval_expr("from opencog.atomspace import *");
+        evaluator->eval_expr("ATOMSPACE = AtomSpace()");
         evaluator->eval_expr("ATOMSPACE.add_node(types.ConceptNode,'test')");
         evaluator->eval_expr("print ('Total atoms = ', ATOMSPACE.size(), '\\n')");
 
-        // Delete the singleton instance and atomspace.
+        // Delete the singleton instance.
         PythonEval::delete_singleton_instance();
-        delete atomSpace;
-        atomSpace = NULL;
         logger().debug("[PythonEvalUTest] testPythonEvalEvalExpr() DONE");
     }
 
     void testApplyAndApplyTV()
     {
-        AtomSpace *as = new AtomSpace();
-        PythonEval::create_singleton_instance(as);
+        PythonEval::create_singleton_instance();
         PythonEval* python = &PythonEval::instance();
 
         // Define python functions for use by scheme.
-        python->eval(
-            "from opencog.atomspace import types, Atom, TruthValue\n"
+        std::string eval_res = python->eval(
+            "from opencog.atomspace import vsfvkdfjnv Atom, TruthValue\n"
```

Comment:
Does this typo demonstrates that this tests never fails? :-)

Reply:
Unintentionally, though it somehow passes